src/shared/components/login/login.js

```diff
@@ -36,7 +36,8 @@ class Login extends Component {
   };
 
   onEmailChange = (value, valid) => {
-    this.setState({ email: value, emailValid: valid });
+    this.props.addEmail(value);
```

Comment:
Firing the Redux action on every keystroke. You probably want to do this when the login button is actually clicked. (`handleOnClick()`)